test/metabase/driver/common/parameters/values_test.clj

```diff
@@ -317,6 +324,61 @@
                      :card-id      (:id card)}
                     []))))))))
 
+  (testing "Persisted Models are substituted"
+    (mt/test-driver :postgres
+      (mt/dataset test-data
+        (mt/with-everything-store
+          (mt/with-temporary-setting-values [:persisted-models-enabled true]
+            (let [mbql-query (mt/mbql-query categories)]
+              (mt/with-temp* [Card [model {:name "model"
+                                           :dataset true
+                                           :dataset_query mbql-query
+                                           :database_id (mt/id)}]]
+                (ddl.i/check-can-persist (mt/db))
+                (persisted-info/ready-database! (mt/id))
+                (#'task.persist-refresh/refresh-tables!
+                 (mt/id)
+                 (var-get #'task.persist-refresh/dispatching-refresher))
```

Comment:
could all this be a `with-persisted-model` that takes in an mbql-query ? or maybe a card-definition?

Reply:
I ended up making a macro that will do the necessary setup and then let you call a function to persist. This is because you might need lots of temp items (collections, other cards) and in one test we need to assert some counts before and after persistence. This also avoids having to teach clj-kondo how to parse model definitions in another place.

Reply:
I like your version. having the `persists-fn` is convienet for cases where we want to test things before the persistent happens.